chore: change required PHP versions and upgrade PestPHP versions

misaert opened the pull request:
Add:
- PHP `8.3`
- PHP `8.4`
- PHP `8.5`

Remove:
- PHP `8.0`
- PHP `8.1`

Keep:
- PHP `8.2`

It also upgrades the PestPHP versions to remove v1 and add v2, v3 and v4 because of this error with PestPHP v1 and PHP `8.4` and `8.5`:
```
  Pest\PendingObjects\TestCall::__construct(): Implicitly marking parameter $description as nullable is deprecated, the explicit nullable type must be used instead

  at vendor/pestphp/pest/src/PendingObjects/TestCall.php:52
     48▕
     49▕     /**
     50▕      * Creates a new instance of a pending test call.
     51▕      */
  ➜  52▕     public function __construct(TestSuite $testSuite, string $filename, string $description = null, Closure $closure = null)
     53▕     {
     54▕         $this->testCaseFactory = new TestCaseFactory($filename, $description, $closure);
     55▕         $this->testSuite       = $testSuite;
     56▕         $this->descriptionLess = $description === null;

      +5 vendor frames
  6   tests/SafeEventsDispatcherTest.php:20
```

It closes #8.

misaert committed:
ci: add PHP versions

- PHP `8.3`
- PHP `8.4`

misaert committed:
ci: add PHP `8.5`

bpolaszek commented:
Depends on:
- https://github.com/matthiasnoback/phpunit-test-service-container/pull/9
- https://github.com/matthiasnoback/doctrine-dbal-test-service-provider/pull/6

misaert committed:
chore: remove unmaintained PHP versions

misaert committed:
chore: remove PHP 8.5 for now

Error with PestPHP v1:
```
  Pest\PendingObjects\TestCall::__construct(): Implicitly marking parameter $description as nullable is deprecated, the explicit nullable type must be used instead

  at vendor/pestphp/pest/src/PendingObjects/TestCall.php:52
     48▕
     49▕     /**
     50▕      * Creates a new instance of a pending test call.
     51▕      */
  ➜  52▕     public function __construct(TestSuite $testSuite, string $filename, string $description = null, Closure $closure = null)
     53▕     {
     54▕         $this->testCaseFactory = new TestCaseFactory($filename, $description, $closure);
     55▕         $this->testSuite       = $testSuite;
     56▕         $this->descriptionLess = $description === null;

      +5 vendor frames
  6   tests/SafeEventsDispatcherTest.php:20
```

misaert committed:
revert: "chore: remove PHP 8.5 for now"

This reverts commit c8a9ab1820c84222993f37fe9205bbc5f4225d9e.

misaert committed:
chore: upgrade PestPHP versions

misaert committed:
style: fix PHPStan error

```
 -- ----------------------------------------------------------------------------------------------------------
     Error
 -- ----------------------------------------------------------------------------------------------------------
     Ignored error pattern #Call to an undefined method Pest\\Expectation\|Pest\\Support\\Extendable# in path
     /home/misaert/dev/repo/github/_forks/doctrine-safe-events/tests/* was not matched in reported errors.
 -- ----------------------------------------------------------------------------------------------------------
```

misaert committed:
style: fix PHPStan error

```
 ------ -----------------------------------------------------------------------------
  Line   tests/SafeEventsDispatcherTest.php
 ------ -----------------------------------------------------------------------------
  22     Call to an undefined method PHPUnit\Framework\TestCase::getEntityManager().
  62     Call to an undefined method PHPUnit\Framework\TestCase::getEntityManager().
  102    Call to an undefined method PHPUnit\Framework\TestCase::getEntityManager().
 ------ -----------------------------------------------------------------------------
```

misaert commented:
Updated PR description.

misaert commented:
Is it ok for you, @bpolaszek ?

Thanks,

Mika

bpolaszek commented:
Thanks a lot @misaert ! Much appreciated 🙏
